prepare-repository.ps1

```diff
@@ -0,0 +1,71 @@
+<#
+.SYNOPSIS
+    Prepare the GoWSL repository so you can run the examples and tests.
+    You will be prompted to install WSL and Ubuntu from the Microsoft store.
+    Winget must be available.
+.PARAMETER AcceptAll
+    Enable this flag to automatically accept package and source agreements for
+    these dependencies. This will make the install non-interactive.
+.EXAMPLE
+    .\prepare-repository.ps1
+
+    Prepare this repo in your machine.
+.EXAMPLE
+    powershell -ExecutionPolicy Bypass .\prepare-repository.ps1
+
+    You may have script execution disallowed. This command will bypass it.
+.EXAMPLE
+    .\prepare-repository.ps1 -AcceptAll
+
+    Prepare this repo non-interactively.
+#>
+
+param (
+    [switch]$AcceptAll = $false
+)
+
+$acceptance = ""
+if ( $AcceptAll ) {
+    $acceptance = "--accept-package-agreements","--accept-source-agreements"
+}
+
+function Test-Winget {
+    &{ winget --version } 2>&1 | Out-Null
+    if ( $LASTEXITCODE -eq "0" ) { return $true }
+    Write-Error "Winget is not installed. Please install it from https://learn.microsoft.com/en-us/windows/package-manager/winget/"
+    return $false
+}
+
+# Gettting WSL
+if ( ! $(Get-AppPackage | Where-Object Name -like 'MicrosoftCorporationII.WindowsSubsystemForLinux') ) {
+    if (! $(Test-Winget) ) { Exit(1) }
+    Write-Output "Installing WSL"
+    winget install --name 'Windows Subsystem for Linux' --silent ${acceptance}
+    if ( ! $? ) { Exit(1) }
+}
+Write-Output "WSL is installed"
+
+# Gettting Ubuntu
+if ( $(Get-AppPackage | Where-Object Name -like 'CanonicalGroupLimited.Ubuntu').Count -eq 0 ) {
+    if (! $(Test-Winget) ) { Exit(1) }
+    Write-Output "Installing Ubuntu"
+    winget install --Id '9PDXGNCFSCZV' --silent ${acceptance}
```

Comment:
I would like as a const (if possible in powershell?) or at least something defined as a global variable on top (easier to spot if we need to change something)

Reply:
I put it at the top of the file (as top as it'll let me) so it's easier to see.